src/main/java/Duke.java

```diff
@@ -1,10 +1,135 @@
+                " What can I do for you? uwu");
+    }
+
+    public static void bye() {
+        willExit = true;
+        printFormat(" Aw, are you leaving now?\n" +
+                " Hope to see you again soon!");
+    }
+
+    public static void list() {
+        if (taskIndex == 0) {
+            printFormat("You haven't added any tasks to your list yet!");
+            return;
+        }
+
+        String listAsString = "";
+        for (int i = 0; i < taskIndex; i++) {
+            Task curr = list[i];
+            listAsString = listAsString.concat(String.format(" %d. %s\n", i + 1, curr.toString()));
+        }
+        printFormat("Here are the tasks in your list:\n" + listAsString);
+    }
+
+    public static void markStatus(Boolean shouldMark, String line) {
+        Task curr;
+        try {
+            int taskNum = Integer.parseInt(line.split(" ", 0)[1]);
+            curr = list[taskNum - 1];
+        } catch (Exception exception) {
+            printFormat("Please mark / unmark with a number that's in the list :')");
+            return;
+        }
+
+        if (shouldMark) {
+            curr.setDone(true);
+            printFormat("Nice! I've marked this task as done:\n  " + curr.toString());
+        } else {
+            curr.setDone(false);
+            printFormat("OK, I've marked this task as not done yet:\n  " + curr.toString());
+        }
+    }
+
+    private static Task parseDeadline(String description) {
+        String[] deadlineBreakdown = description.split("/by", 2);
+        description = deadlineBreakdown[0];
+        String by = deadlineBreakdown[1];
+        Task t = new Deadline(description, by);
+        return t;
+    }
+
+    private static Task parseEvent(String description) {
+        String[] eventBreakdown = description.split(" /at ", 2);
+        description = eventBreakdown[0];
+        String at = eventBreakdown[1];
+        Task t = new Event(description, at);
+        return t;
+    }
+
+    private static Task parseTask(String type, String description) {
+        Task t;
+        if (type.equals("todo")) {
+            t = new Todo(description);
+        } else if (type.equals("deadline")) {
+            t = parseDeadline(description);
+        } else if (type.equals("event")) {
+            t = parseEvent(description);
+        } else {
+            throw new RuntimeException("Not a valid task type");
+        }
+        return t;
+    }
+
+    public static void invalidTask() {
+        printFormat("I don't understand what you want to do.\n" +
+                "Maybe you could try the following commands:\n" +
+                "  - list: list out existing tasks\n" +
+                "  - etc.");
+    }
+
+    public static void addTask(String line) {
+        try {
+            String[] commands = line.split(" ", 2);
+            String type = commands[0];
+            String description = commands[1];
+
+            Task t = parseTask(type, description);
+            list[taskIndex] = t;
```

Comment:
Like Clarence mentioned, I think separating parseTask from addTask is a good way to improve code readbility. Definitely reduces the number of lines in a single method and not as painful to the eye!